compiler+runtime/bin/jank/compiler+runtime/bash_test.clj

```diff
@@ -25,19 +26,25 @@
           (if skip?
             (util/log-warning "Skipped " relative-dirname)
             (util/with-elapsed-time duration
-              (let [res @(b.p/process {:out :string
-                                       :err :out
-                                       :dir dirname
-                                       :extra-env extra-env}
-                                      test-file)]
-                (when (or (and (zero? (:exit res)) expect-failure?)
+              (let [p (b.p/process {:out :string
+                                    :err :out
+                                    :dir dirname
+                                    :extra-env extra-env}
+                                   test-file)
+                    res (deref p (* timeout-seconds-per-script 1000) :timeout)]
+                (when (or (= :timeout res)
+                          (and (zero? (:exit res)) expect-failure?)
                           (and (not (zero? (:exit res))) (not expect-failure?)))
-                  (vreset! unexpected-result res)))
+                  (case res
+                    :timeout (do (b.p/destroy-tree p)
+                                 (vreset! unexpected-result (assoc @p :timeout true)))
```

Comment:
Please always put a new line after `do` (below, on line 43, too). Also, please use `:timeout?` since we're talking about a bool.